### PR TITLE
add audio language to stream data

### PR DESCRIPTION
**Pull request by herby2212**
## Description

Addition of the audio language to the stream data, so it can be viewed in the player details (stream_data.html) in the history.
For previous history items this field in the details is not displayed.

get_stream_data API method will now also return audio_language and stream_audio_language. See updated json structure example.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods

### CODE REVIEW

File: plexpy/__init__.py [hunk 1]
```diff
@@ -613,13 +613,13 @@ def dbcheck():
...
         'video_dynamic_range TEXT, aspect_ratio TEXT, '
-        'audio_codec TEXT, audio_bitrate INTEGER, audio_channels INTEGER, subtitle_codec TEXT, '
+        'audio_codec TEXT, audio_bitrate INTEGER, audio_channels INTEGER, audio_language TEXT, subtitle_codec TEXT, '
         'stream_bitrate INTEGER, stream_video_resolution TEXT, quality_profile TEXT, '
         'stream_container_decision TEXT, stream_container TEXT, '
         'stream_video_decision TEXT, stream_video_codec TEXT, stream_video_bitrate INTEGER, stream_video_width INTEGER, '
         'stream_video_height INTEGER, stream_video_framerate TEXT, stream_video_scan_type TEXT, stream_video_full_resolution TEXT, '
         'stream_video_dynamic_range TEXT, '
-        'stream_audio_decision TEXT, stream_audio_codec TEXT, stream_audio_bitrate INTEGER, stream_audio_channels INTEGER, '
+        'stream_audio_decision TEXT, stream_audio_codec TEXT, stream_audio_bitrate INTEGER, stream_audio_channels INTEGER, stream_audio_language TEXT, '
```
Comment (JonnyWong16) [inline plexpy/__init__.py]: Break to a new line to keep the line length down.

File: plexpy/__init__.py [hunk 2]
```diff
@@ -666,7 +666,7 @@ def dbcheck():
         'stream_video_decision TEXT, stream_video_bitrate INTEGER, stream_video_codec TEXT, stream_video_codec_level TEXT, '
         'stream_video_bit_depth INTEGER, stream_video_height INTEGER, stream_video_width INTEGER, stream_video_resolution TEXT, '
         'stream_video_framerate TEXT, stream_video_scan_type TEXT, stream_video_full_resolution TEXT, stream_video_dynamic_range TEXT, '
-        'stream_audio_decision TEXT, stream_audio_codec TEXT, stream_audio_bitrate INTEGER, stream_audio_channels INTEGER, '
+        'stream_audio_decision TEXT, stream_audio_codec TEXT, stream_audio_bitrate INTEGER, stream_audio_channels INTEGER, stream_audio_language TEXT, '
```
Comment (JonnyWong16) [inline plexpy/__init__.py]: Break to a new line to keep the line length down.

File: plexpy/__init__.py [hunk 3]
```diff
@@ -1711,7 +1723,19 @@ def dbcheck():
...
         )
-
+    
+    # Upgrade session_history_media_info table from earlier versions
+    try:
+        c_db.execute('SELECT audio_language FROM session_history_media_info')
+    except sqlite3.OperationalError:
+        logger.debug("Altering database. Updating database table session_history_media_info.")
+        c_db.execute(
+            'ALTER TABLE session_history_media_info ADD COLUMN audio_language TEXT '
```
Comment (JonnyWong16) [inline plexpy/__init__.py]: Remove extra space at the end of the string.

File: plexpy/__init__.py [hunk 4]
```diff
@@ -1711,7 +1723,19 @@ def dbcheck():
...
+        )
+        c_db.execute(
+            'ALTER TABLE session_history_media_info ADD COLUMN stream_audio_language TEXT '
```
Comment (JonnyWong16) [inline plexpy/__init__.py]: Remove extra space at the end of the string.

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]
We should save both the `audio_language` and `audio_language_code`.

For example, Japanese:
```
audio_language = "日本語"
audio_language_code = "jpn"
```

But still only display the `language` in the stream details.

**Comment by herby2212** [comment]
Sounds good. (I originally used the language code, but then opted for the language name to be displayed). I will make the changes and then push the commit.

**Review by JonnyWong16 — APPROVED** [review]

**Comment by JonnyWong16** [comment]
Looks good. Thanks.

**Pull request merged**